3rdparty/halrvv/rvv071/halrvv071.hpp

```diff
@@ -0,0 +1,149 @@
+
+#ifndef CV_HAL_RVV_071_ENABLED
+#  if defined(__THEAD_VERSION__) && defined(__riscv_v) && __riscv_v == 7000
+#    define CV_HAL_RVV_071_ENABLED 1
+#  else
+#    define CV_HAL_RVV_071_ENABLED 0
+#  endif
+#endif
+
+#endif /* GCC version check*/
+
+#if CV_HAL_RVV_071_ENABLED
+
+#include "opencv2/core/hal/interface.h"
+
+#include <riscv_vector.h>
+
+#undef cv_hal_cvtBGRtoBGR
+#define cv_hal_cvtBGRtoBGR cvt_hal_BGRtoBGR
+
+static unsigned char index_array_32 [32] 
+                        { 2, 1, 0, 3, 6, 5, 4, 7, 10, 9, 8, 11, 14, 13, 12, 15, 18, 17, 16, 19, 22, 21, 20, 23, 26, 25, 24, 27, 30, 29, 28, 31  };
+
+static unsigned char index_array_24 [24]
+                        { 2, 1, 0, 5, 4, 3, 8, 7, 6, 11, 10, 9, 14, 13, 12, 17, 16, 15, 20, 19, 18, 23, 22, 21  };
+
+static void cvt_vector(const uchar* src, uchar * dst, uchar * index, int n, int scn, int dcn, int bi, int vsize_pixels, int vsize)
+{
+    vuint8m2_t vec_index = vle8_v_u8m2(index, vsize);
+
+    int i = 0;
+
+    for( ; i <= n-vsize; i += vsize_pixels, src += vsize, dst += vsize)
+    {
+        vuint8m2_t vec_src = vle8_v_u8m2(src, vsize);
+        vuint8m2_t vec_dst = vrgather_vv_u8m2(vec_src, vec_index, vsize);
+        vse8_v_u8m2(dst, vec_dst, vsize);
+    }
+
+    for ( ; i < n; i++, src += scn, dst += dcn )
+    {
+        uchar t0 = src[0], t1 = src[1], t2 = src[2];
+        dst[bi  ] = t0;
+        dst[1]    = t1;
+        dst[bi^2] = t2;
+        if(dcn == 4)
+        {
+            uchar d = scn == 4 ? src[3] : UCHAR_MAX;
+            dst[3] = d;
+        }
+    }
+}
+
+static void cvt_scalar(const uchar* src, uchar * dst, int n, int scn, int dcn, int bi)
+{
+    for (int i = 0; i < n; i++, src += scn, dst += dcn)
+    {
+        uchar t0 = src[0], t1 = src[1], t2 = src[2];
+        dst[bi  ] = t0;
+        dst[1]    = t1;
+        dst[bi^2] = t2;
+        if(dcn == 4)
+        {
+            uchar d = scn == 4 ? src[3] : UCHAR_MAX;
+            dst[3] = d;
+        }
+    }
+}
+
+static int cvt_hal_BGRtoBGR(const uchar * src_data, size_t src_step, uchar * dst_data, size_t dst_step, int width, int height, int depth, int scn, int dcn, bool swapBlue)
+{
+    int blueIdx = swapBlue ? 2 : 0;
+    if (scn == dcn)
+    {   
+        const int vsize_pixels = 8;
+        const int vsize = vsize_pixels*scn;
+    
+        unsigned char* index;
+        if (scn == 4)
+        {
+            index = index_array_32;
+        }
+        else
+        {
+            index = index_array_24;
+        }
+
+        size_t vl = vsetvl_e8m2(vsize);
+
+        for(int i = 0; i < height; i++, src_data += src_step, dst_data += dst_step)
+            cvt_vector(src_data, dst_data, index, width, scn, dcn, blueIdx, vsize_pixels, vsize);
+    }
+    else
+    {
+        for(int i = 0; i < height; i++, src_data += src_step, dst_data += dst_step)
+            cvt_scalar(src_data, dst_data, width, scn, dcn, blueIdx);
+    }
+
+    return CV_HAL_ERROR_OK;
+}
+#endif // CV_HAL_RVV_071_ENABLED
+
+#endif
```

Comment:
Missing empty line at the end of file. Try to install or enable EditorConfig file support for your editor, it would handle such nuances automatically, according to our `.editorconfig` file.